routing/pathfind.go

```diff
@@ -902,7 +913,15 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		// size is zero, because the original htlc isn't in the onion
 		// blob.
 		var payloadSize uint64
-		if fromVertex != source {
+		ToPubKey := edge.policy.ToNodePubKey()
+
+		// If the edge points to the NUMS key we don't need to consider
+		// the payload size again because we already added the size of
+		// the dummy hop which will be removed after the route is found.
+		// The dummy target hop has the same cipher text length as the
+		// real last blinded target hop.
+		if fromVertex != source &&
+			!isBlindedRouteNUMSTargetKey(ToPubKey[:]) {
```

Comment:
is there not something more hidden we can do where the `edge.hopPayloadSizeFn` returns 0 for the dummy hop edge?